src/HealthChecks.SendGrid/DependencyInjection/SendGridHealthCheckExtensions.cs

```diff
@@ -31,13 +31,41 @@ public static IHealthChecksBuilder AddSendGrid(
         IEnumerable<string>? tags = default,
         TimeSpan? timeout = default)
     {
-        var registrationName = name ?? NAME;
+        return builder.AddSendGrid(_ => apiKey, name, failureStatus, tags, timeout);
```

Comment:
nit: we are now capturing a closure (so there is a performance) penalty, but it helps to reduce code duplication, so I am fine with that. We may however, change that approach in the future.